### DISCUSSION TITLE
Hardware Migration

### DISCUSSION THREAD

**Discussion by Voz-DeCarabas** [lavinir/hassio-onedrive-backup#83] [General]
Hi all.  I'm running the backups to onedrive and all working well, although I guess I wouldn't know how well until needing to restore!

A quick question - could the recovery process be used to help with migration from one hardware setup to another?  Once I can find and buy one I plan to get a raspberry pi 4 and then look at migrating from my current pi 3 but attempt to keep the setup the same.  I found a post from a couple of years ago that gives a process as below using the backups and rather than using Samba etc could I connect the new hardware to OneDrive and recover from there?

- Download the Hass.io image for Rpi4
- Image the new microSD card
- Let the the Rpi4 boot and sit and download everything
- Make a snapshot on the Rpi3 and download it
- On the Rpi4 install Samba Share and copy the snapshot into the backup folder
- Reboot the Rpi4
- Restore the snapshot on the Rpi4
- Shutdown the Rpi3 and change the Rpi4 to have the IP of the Rpi3

**Comment by lavinir** [comment]
Hi,

Of course you can use the backups to migrate to new hardware. This functionality is already present inside the Home Assistant backup solution. The convenience you get with this addon is the integration with the cloud.
All you would need to do here is
1. Load up a new instance of Home Assistant on your new hardware
2. Install the OneDrive Backup Addon (with the same Microsoft Account)
3. You should be able to view the back ups in OneDrive from the Addon web interface. 
4. Choose and download the backups you want to restore. 
5. Navigate to the Backups section in Home Assistant and restore it

**Reply by Voz-DeCarabas** [reply]
Thanks for confirming my assumptions.  The last time I personally did a restore from backup at all was many years ago and we had all sorts of issues where the hardware wasn't identicle. Thankfully, things get better :-)
